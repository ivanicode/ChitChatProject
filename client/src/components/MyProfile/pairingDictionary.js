export const pairingRelationship = {
    1: 1,
    2: 2,
    3: null
}