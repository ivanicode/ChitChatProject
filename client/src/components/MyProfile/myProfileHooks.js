import {hobby} from '../RegisterProfile/dictionary';
import { useFetch } from '../../common/hooks/useFetchHook';
import { useState, useEffect } from 'react'
import {pairingRelationship} from './pairingDictionary'
import { useGetMyContext } from '../Home/Home'

export function useMyProfileHooks() {
    const [hobbys, setHobbys] = useState([])
    const [age, setAge] = useState()
    const {data: accountData} = useFetch('/api/user')
    const {data} = useFetch('/api/user/details');
    const [shouldFetchMatch, setShouldFetchMatch] = useState(false)
    const [fetchMatchUrl, setFetchMatchUrl] = useState('')
    const {data: matchData} = useFetch(fetchMatchUrl, shouldFetchMatch);
    const { updatePartnerData } = useGetMyContext()

    useEffect(() => {
        if(data?.interests){
            const chosenHobbys = data?.interests;
            setHobbys(hobby.filter((element) => chosenHobbys.includes(element.id)).map(el => el?.label))
        }
        if(data?.relationship){
            setFetchMatchUrl(`/api/user/?relationship=${pairingRelationship[data.relationship]}`)
        }
    }, [data])

    useEffect(() => {
        if(accountData?.birth_date){
            const now = new Date();
            const birthDate = new Date(accountData.birth_date)
            setAge(now.getFullYear() - birthDate.getFullYear())
        }
    }, [accountData])

    useEffect(() => {
        if(matchData?.id){
            updatePartnerData(matchData)
        }
    }, [matchData])

    function findMatch(event){
        setShouldFetchMatch(true)
    }
    

    return {data, hobbys, age, findMatch, matchData}
}