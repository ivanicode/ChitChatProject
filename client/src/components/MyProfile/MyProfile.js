import React from 'react';
import {useMyProfileHooks} from './myProfileHooks'


export function MyProfile () {

    
const {data, hobbys, age} = useMyProfileHooks()

    return (
        <div className="myProfile">
            <div className="mainPicture">
                
            </div>
            <div className="statusNickname">
                <div className="nickname">
                    {data?.nickname}
                </div>
                <div className="status">
                    <img src={'/' + data?.relationship + '.png'} className="statusImg"></img>
                </div>
            </div>
            <div className="interests">               
                <div className="city">{data?.city}</div>
                <div className="age">{age}</div>
                <div className="hobbys">{hobbys.join(', ')}</div>
            </div>
            <div className="myPage">
                
            </div>
        </div>
    );   
}

export default MyProfile;
