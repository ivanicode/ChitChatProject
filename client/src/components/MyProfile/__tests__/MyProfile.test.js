import React from 'react';
import {shallow} from 'enzyme';
import {MyProfile} from '../MyProfile';


describe('MyProfile component', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(<MyProfile />)
        expect(wrapper).toMatchSnapshot()
    })
})