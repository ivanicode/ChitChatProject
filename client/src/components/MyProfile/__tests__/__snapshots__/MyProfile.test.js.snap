// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile component should match snapshot 1`] = `
<div
  className="myProfile"
>
  <div
    className="mainPicture"
  />
  <div
    className="statusNickname"
  >
    <div
      className="nickname"
    />
    <div
      className="status"
    />
  </div>
  <div
    className="interests"
  >
    <div
      className="city"
    />
    <div
      className="age"
    />
    <div
      className="hobbys"
    />
  </div>
  <div
    className="myPage"
  />
</div>
`;
